### PR TITLE
Add mock for stx-account in pox-3

### PR DESCRIPTION
**Pull request by friedger**
This PR
* adds a setter and getter functions for the unlocked, locked amount and unlock-height for any users in pox-3
* replaces `stx-account` with `get-stx-account` in pox-3
* see #549

### CODE REVIEW

**Comment by friedger** [comment]
Tests will be added when the new test generation has landed.

**Comment by friedger** [comment]
Blocked by https://github.com/Trust-Machines/stacks-sbtc/issues/582

File: sbtc-mini/ext/generate-tests.ts [hunk 1]
```diff
@@ -93,9 +93,14 @@ function generatePrepareTx(contractPrincipal: string, annotations: FunctionAnnot
...
 }
 
+function generatePox3AllowContractCaller(annotations: FunctionAnnotations) {
+	return `Tx.contractCall(deployer.address + '.pox-3', 'allow-contract-caller', [ "'" + deployer.address + '${annotations['poxadmin']}', 'none'], callerAddress)`;
+}
+
 function generateNormalMineBlock(contractPrincipal: string, testFunction: string, annotations: FunctionAnnotations) {
 	return `let block = chain.mineBlock([
 		${annotations['prepare'] ? `${generatePrepareTx(contractPrincipal, annotations)},` : ''}
+		${annotations['poxadmin'] ? `${generatePox3AllowContractCaller(annotations)},` : ''}
```
Comment (MarvinJanssen) [inline sbtc-mini/ext/generate-tests.ts:103]: Nice solution but what I do not like is that it makes the generation script more opinionated. How would you feel about an annotation that calls a custom TS function? (Just how `bootstrap.ts` is separated now.)

File: sbtc-mini/contracts/pox-3.clar [hunk 2]
```diff
@@ -224,6 +224,19 @@
...
 
+;; MOCK
+;; Allow to set stx-account details for any user
+;; These values are used for PoX only
+(define-map stx-account-details principal {unlocked: uint, locked: uint, unlock-height: uint})
+
+(define-read-only (get-stx-account (user principal))
+    (default-to (stx-account user) (map-get? stx-account-details user)))
+
+(define-public (set-stx-account (user principal) (details {unlocked: uint, locked: uint, unlock-height: uint}))
```
Comment (MarvinJanssen) [inline sbtc-mini/contracts/pox-3.clar:235]: Can we prefix these functions names with `mock-` or `debug-`? I prefer the latter I think. It is something like that in `clarity-bitcoin` too.

**Review by MarvinJanssen — COMMENTED** [review]
Did a quick pass. Will do a full review when you are unblocked and can write all tests.

File: sbtc-mini/ext/generate-tests.ts (hunk 1 above)
Comment (friedger) [reply sbtc-mini/ext/generate-tests.ts:103]: This should not be part of the PR. New PR sent with changes only to pox-3.

File: sbtc-mini/contracts/pox-3.clar (hunk 2 above)
Comment (friedger) [reply sbtc-mini/contracts/pox-3.clar:235]: adding `mock-`

**Comment by friedger** [comment]
Replaced by #586

**Pull request closed without merging**